src/processors/encounter/encounter-processor-helper.ts

```diff
@@ -332,3 +374,31 @@ export async function hydrateRepeatField(
     }),
   ).then((results) => results.flat());
 }
+
+function prepareDiagnosis(fields: FormField[]) {
+  return fields
+    .filter((field) => field.type === 'diagnosis')
+    .map((field) => {
+      if (field?.meta?.repeat?.wasDeleted) {
+        const voided = {
+          uuid: field?.meta?.previousValue?.uuid,
+          voided: true,
+        };
+        if (field.meta?.submission) {
+          field.meta.submission.voidedValue = voided;
+        }
+        return field;
+      }
+      return field;
```

Comment:
Move this logic to the diagnosis adapter. It's responsible for voiding deleted objects. At this point, we're supposed to collect available submissions. The function can be simplified:

```ts
const diagnoses = fields.filter((field) => field.type === 'diagnosis' && hasSubmission(field))
                      .map((field) => field.meta.submission.newValue || field.meta.submission. voidedValue)

```

Reply:
The diagnosis adapter loads before the deletedFields are updated in the component, thats why i did this here. If you log the value of deletedFields in the diagnosis adapter, it will be empty. Same reason i couldn't use the `wasDeleted `

Reply:
Refer to my comment: https://github.com/openmrs/openmrs-esm-form-engine-lib/pull/298#discussion_r1913312779